pkg/types/entries.go

```diff
@@ -36,11 +37,23 @@ type EntryImpl interface {
 	Unmarshal(e models.ProposedEntry) error           // unmarshal the abstract entry into the specific struct for this versioned type
 	Validate() error                                  // performs any cross-field validation that is not expressed in the OpenAPI spec
 	Attestation() (string, []byte)
+	CreateFromPFlags(context.Context, ArtifactProperties) (models.ProposedEntry, error)
```

Comment:
nit: Should this name still reference `PFlags`, since you've already done the translate into ArtifactProperties?